apps/web/hooks/useS3.tsx

```diff
@@ -63,11 +63,22 @@ export const S3Provider: React.FC<PropsWithChildren<Props>> = ({
 		isMounted.current = true;
 		if (data.keys.bucketUrl) return;
 
-		if ((Provider[data.type] as Provider) === Provider.s3) {
-			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.amazonaws.com`;
-		} else if ((Provider[data.type] as Provider) === Provider.backblaze) {
-			data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.backblazeb2.com`;
-		}
+		switch (Provider[data.type] as Provider) {
+			case Provider.s3:
+			  data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.amazonaws.com`;
+			  break;
+			case Provider.backblaze:
+			  data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.backblazeb2.com`;
+			  break;
+			case Provider.wasabi:
+			  data.keys.bucketUrl = `https://${data.keys.Bucket}.s3.${data.keys.region}.wasabisys.com`;
+			  break;
+			case Provider.digitalocean:
+			  data.keys.bucketUrl = `https://${data.keys.Bucket}.${data.keys.region}.digitaloceanspaces.com`;
+			  break;
+			default:
+			  break;
+		  }
```

Comment:
@9brian https://github.com/faisalsayed10/firefiles/commit/b7515e1f79ba112b7ca56cb7d107f579af4e1fc0#r122804531 can you address my comments here?